src/braket/default_simulator/density_matrix_simulator.py

```diff
@@ -120,9 +213,21 @@ def properties(self) -> GateModelSimulatorDeviceCapabilities:
                             {"name": "Probability", "minShots": 0, "maxShots": max_shots},
                             {"name": "DensityMatrix", "minShots": 0, "maxShots": 0},
                         ],
-                    }
+                    },
                 },
                 "paradigm": {"qubitCount": qubit_count},
                 "deviceParameters": GateModelSimulatorDeviceParameters.schema(),
             }
         )
+
+    @singledispatchmethod
+    def _run_internal(
+        self, ir: Union[OQ3Program, JaqcdProgram], *args, **kwargs
+    ) -> GateModelTaskResult:
+        """run OpenQASM program"""
+        return OpenQASMDensityMatrixSimulator.run(self, ir, *args, **kwargs)
```

Comment:
I'm a bit confused; why is it necessary to subclass both `OpenQASMDensityMatrixSimulator` and `JaqcdDensityMatrixSimulator` if `run` just does this dispatch?

Reply:
because we are passing in `self` to the `OpenQASMDensityMatrixSimulator` method. We can't do this if `self` isn't an instance of `OpenQASMDensityMatrixSimulator`